packages/workspace/src/browser/workspace-commands.ts

```diff
@@ -323,6 +362,26 @@ export class WorkspaceCommandContribution implements CommandContribution {
         return new WorkspaceRootUriAwareCommandHandler(this.workspaceService, this.selectionService, handler);
     }
 
+    protected handleCopy(text: string | undefined): void {
+        if (!text) {
+            return;
+        }
+        if (document.documentElement) {
+            const toDispose = addClipboardListener(document.documentElement, 'copy', event => {
+                toDispose.dispose();
+                this.addToClipboard(event, text);
+            });
+            document.execCommand('copy');
+        }
+    }
+
+    protected addToClipboard(event: ClipboardEvent, text: string): void {
```

Comment:
We should extract `ClipboardService` from https://github.com/theia-ide/theia/pull/5527, make it work on as many browser as we can and use it in all such places instead of implementing different approaches each time.